volatility3/framework/plugins/linux/iomem.py

```diff
@@ -0,0 +1,156 @@
+        end = format_hints.Hex(resource.end)
+
+        # get name with protection against smear as following a pointer
+        try:
+            name = utility.pointer_to_string(resource.name, 128)
+        except exceptions.InvalidAddressException:
+            vollog.warning(
+                "Unable to follow pointer to name for resource object at {resource_offset:#x}, "
+                "replaced with UnreadableValue"
+            )
+            name = renderers.UnreadableValue()
+
+        # mark this resource as seen in the seen set. Normally this should not be needed but will protect
+        # against possible infinite loops. Warn the user if an infinite loop would have happened.
+        if resource_offset in seen:
+            vollog.warning(
+                f"The resource object at {resource_offset:#x} '{name}' has already been processed, "
+                "this should not normally occur. No further results from related resources will be "
+                "displayed to protect against infinite loops."
+            )
+            return None
+        else:
+            seen.add(resource_offset)
+
+        # yield information on this resource
+        yield depth, (name, start, end)
+
+        # process child resource if this exists
+        if resource.child != 0:
+            yield from cls.parse_resource(
+                context,
+                vmlinux_module_name,
+                resource.child,
+                seen,
+                depth + 1,
+            )
+
+        # process sibling resource if this exists
+        if resource.sibling != 0:
+            yield from cls.parse_resource(
+                context,
+                vmlinux_module_name,
+                resource.sibling,
+                seen,
+                depth,
+            )
+
+    def _generator(self):
+        """Generates an output similar to /proc/iomem on a running system
+
+        Args:
+            None
+
+        Yields:
+            Each row of output using the parse_resource function
+        """
+
+        # get the kernel module from the current context
+        vmlinux_module_name = self.config["kernel"]
+        vmlinux = self.context.modules[vmlinux_module_name]
+
+        # check that the iomem_resource symbol exists
+        # normally exported in /kernel/resource.c
+        try:
+            iomem_root_offset = vmlinux.get_absolute_symbol_address("iomem_resource")
+        except exceptions.SymbolError:
+            iomem_root_offset = None
+
+        # error if 'iomem_resource' is not found
+        if not iomem_root_offset:
+            raise TypeError(
+                "This plugin requires the iomem_resource structure. This structure is not present in the supplied symbol table. This means you are either analyzing an unsupported kernel version or that your symbol table is corrupt."
+            )
+
+        # error if type 'resource' is not found
+        if not vmlinux.has_type("resource"):
```

Comment:
I might be tempted to do this at the start of run.  It really doesn't make much of a difference, but it's not expected that other plugins would ever call this generator and it feels like "can this plugin run" checks should happen at the start `run` rather than in the generator function?  Happy either way, but figured I'd mention it.  There's so little else in this commit that needs mentioning, really good work!  5:)

Reply:
This is the only comment left here. If you are okay with the checks where they are then lets merge it.

Reply:
I can change where those checks happen, thanks for the reminder.

Reply:
Ah, looks like I already moved these checks into run. I think this is sorted now?